### PR TITLE
Created types for react-pivottable

### PR DESCRIPTION
**Pull request by jbovone**
Please fill in this template.

- [x] Use a meaningful title for the pull request. Include the name of the package modified.
- [x] Test the change in your own code. (Compile and run.)
- [x] [Add or edit tests](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#my-package-teststs) to reflect the change.
- [x] Follow the advice from the [readme](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#make-a-pull-request).
- [x] Avoid [common mistakes](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#common-mistakes).
- [x] [Run `npm test <package to test>`](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#running-tests).

Select one of these and delete the others:

If adding a new definition:
- [x] The package does not already provide its own types, or cannot have its `.d.ts` files generated via `--declaration`
- [x] If this is for an npm package, match the name. If not, do not conflict with the name of an npm package.
- [?] Create it with `dts-gen --dt`, not by basing it on an existing project.
- [x] Represents shape of module/library [correctly](https://www.typescriptlang.org/docs/handbook/declaration-files/library-structures.html)
- [x] `tslint.json` [should contain](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#linter-tslintjson) `{ "extends": "@definitelytyped/dtslint/dt.json" }`, and no additional rules.
- [x] `tsconfig.json` [should have](https://github.com/DefinitelyTyped/DefinitelyTyped/blob/master/README.md#tsconfigjson) `noImplicitAny`, `noImplicitThis`, `strictNullChecks`, and `strictFunctionTypes` set to `true`.

### CODE REVIEW

**Comment by DangerBotOSS** [comment]
Inspecting the JavaScript source for this package found some properties that are not in the .d.ts files.
The check for missing properties isn't always right, so take this list as advice, not a requirement.
## react-pivottable ([<kbd>unpkg</kbd>](https://unpkg.com/browse/react-pivottable@latest/))
was missing the following properties:
1. Dropdown
1. DraggableAttribute
<p align="right">
  Generated by :no_entry_sign: <a href="https://danger.systems/js">dangerJS</a> against 7ae31700dc3364e40ebed84dede422eedd5f708b
</p>

**Comment by peterblazejewicz** [comment]
Hi, 
I think the mdule has no default export. It has ES/CJS/UMD exports, so it can be use in ES/CJS or as global script, but none of the original module exports default from main module. Doing this could lead to errors:
```ts
import PaymentInputs from 'react-pivottable';

console.log(typeof PaymentInputs);
```

Why is the NPM package name different from this module directory? (which translates to import path)

**Comment by jbovone** [comment]
Yes, thanks, this is wrong. 

After manually install my @types it seems that the @types needs to mirror whatever is in the source repo and not only copycat whatever was in the  @type above me  but... yeah, well, it haps... Logical enough :). 
  
I don't know if i should close this. I will open another one.

**Comment by jbovone** [comment]
Ok, I had some problems with the test, but now is ok.
Peter, I improved the stuff, check it up!

File: types/react-pivottable/index.d.ts [hunk 1]
```diff
@@ -0,0 +1,11 @@
+// Type definitions for react-pivottable 0.11
+// Project: https://github.com/medipass/react-payment-inputs
```
Comment (peterblazejewicz) [inline types/react-pivottable/index.d.ts:2]: what is the module project page?

Comment (jbovone) [reply types/react-pivottable/index.d.ts:2]: ohhh... my god!!
https://www.npmjs.com/package/react-pivottable
That was so careless.

**Comment by jbovone** [comment]
Done! 
Peter, thank you. 
If you find more stuff, like in the types that is not best practices, please tell me.

**Review by andrewbranch — CHANGES_REQUESTED** [review]
> I think the mdule has no default export.

It does have a default export 🙈

Check the bottom line: https://unpkg.com/browse/react-pivottable@0.11.0/PivotTableUI.js

**Comment by peterblazejewicz** [comment]
@jbovone nearly there. I was wrong about default export, so your code now in the index should accomodate, something like:
```ts
export { TableInput } from './input';
export { PivotTableProps } from './PivotTable';
import PivotTableUI from './PivotTableUI';
export {
    PivotTableUIProps,
    DraggableAttribute,
    DraggableAttributeProps,
    Dropdown,
    DropdownProps,
} from './PivotTableUI';
export { TQuerySort, TSorterFn, Pivot } from './Utilities';
export { BuildInPlotlyRenderersNames } from './PlotlyRenderers';
export { BuildInRenderersNames, TableRenderers } from './TableRenderers';
export default PivotTableUI;
```

**Comment by jbovone** [comment]
Ok ok, so I was unsure about this, cause his entry-point in the package.json was PivotTableUI. So I include it there. 

like so: 
`import PivotTableUI from 'react-pivottable/PivotTableUI'`

But this one should also work, interesting.
`import PivotTableUI from 'react-pivottable'`

So both ways should work...

**Comment by jbovone** [comment]
Done!

**Comment by jbovone** [comment]
(t's fun how brain squeezers this import mappings can be sometimes...) :brain:

**Review by peterblazejewicz — APPROVED** [review]
LGTM!
@jbovone thanks!

**Review by andrewbranch — APPROVED** [review]

**Comment by jbovone** [comment]
Ok, we are ready.
( ... This stuff will actually work (?) )

**Comment by jbovone** [comment]
Ready to merge

**Pull request merged**